### DISCUSSION TITLE
Installation Process - Issue With './setup.sh' Step

### DISCUSSION THREAD

**Discussion by ValiantEagleSec** [danielmiessler/Fabric#142] [Q&A] [answered]
I will preempt this with I am still fairly new to Linux, Python, and GitHub, but I love this project and want to add it to my learnings.

I am reaching an issue at the step where you use the './setup.sh' command. Note I am using Bash on an Ubuntu VM. Here is the output:

![2024-02-28 12_26_15-Ubuntu  Running  - Oracle VM VirtualBox](https://github.com/danielmiessler/fabric/assets/156213941/0a155103-7862-4c92-ba9b-34e9fe906c8e)

Could someone help me understand the error, and how to resolve it, so I can get Fabric working on my system?

Thank you in advance.

**Comment by davejpeters** [comment]
I'm having the same problem on Debian. Any help would be appreciated.

**Reply by ValiantEagleSec** [reply]
Hey! Check the replies from headbug, they worked for me.

**Comment by headbug** [comment, marked as answer]
Hello,
I had this error when using poetry that is included in my repository, i.e I had the version 1.1.12 from using "sudo apt install python3-poetry".

When I instead used the curl command I got the latest, 1.8.1, which supports the "group" parameter.

` curl -sSL https://install.python-poetry.org | python3 -`

You also have to include the 

`export PATH="/home/yourusername/.local/bin:$PATH"`

 in your ~/.bashrc and then restart your terminal or run "source ~/.bashrc".

Also note that because of the poetry being an old version, now your ~/.bashrc has bad lines at the end, from the error outputs from the setupscript, so make sure to delete them as well.

As a finishing touch, to avoid conflicts, I also deleted the system wide install of poetry, i.e.

`sudo apt remove python3-poetry`

Hope this helps!

**Reply by ValiantEagleSec** [reply]
Hey thank you! Could you please explain how to go about deleting the bad lines at the end of bashrc? I think I got everything else taken care of, but since I'm getting a 

RuntimeError: command not found
Command 'The' not found, did you mean:...

I assume that's because of those bad lines. As I said, still new to Linux but diving in like this is helpful, so any guidance you can give would be appreciated.

**Reply by headbug** [reply]
Sure, but since I have already deleted I don't remember exactly how it looked, but I hope you are able to find the offending text =)

I used a text editor called nano, but any editor will do, vi, vim, emacs, or even code (visual studio code).

So open your .bashrc like so:

`nano ~/.bashrc`

and use the arrows to go to the bottom, where you will find text that (to me at least) is apparent to be error outputs from the faulty poetry run earlier. It was at least 10 rows of text, maybe more.

Ctrl - O to save file and Ctrl - X to exit

**Reply by ValiantEagleSec** [reply]
Awesome, thank you for the help!

**Reply by ValiantEagleSec** [reply]
Update, this did work! I'm up and running, thank you! I had 3 errors to delete from bashrc since I tried multiple times, but once I did that it was smooth sailing.

**Comment by paupier** [comment]
I'm having similar problems installing on Ubuntu 24.04.04 LTS.
I've tried several different install routes and added changes to ~/.bashrc as per @headbug.
![image](https://github.com/danielmiessler/fabric/assets/51230858/eb23ae3e-afbc-4339-9bde-76cbfbea899b)

I just keep getting different versions of the above image.
all resulting in the same output when I get to fabric --setup
![image](https://github.com/danielmiessler/fabric/assets/51230858/25c20b2a-0e16-4782-9975-771200ce36b2)

 been going around in circles... any help would be appreciated. Thanks in advance!

**Comment by smrki1** [comment]
I had the same issue on a mac, pretty much what the problem was for me was that the setup.sh was in another directory. cd to that directory and then running the command was all it took. Also I asked chatgpt to troubleshoot which was quite helpful.

**Reply by paupier** [reply]
Thanks for the chatGPT tips for debugging :)...
Seems like some new instructions came out using pipx install . and that worked almost straight away.
